### PR TITLE
feat: add option to disable the creation of new, empty redirects (fixes: #18)

### PR DESCRIPTION
**Pull request by jeboehm**

### CODE REVIEW

**Comment by tinect** [comment]
Thank you. I like this! 👍 

As discussed in slack, would you like to migrate the config-request into the subscriber and forward its value into the message?

**Review by tinect — CHANGES_REQUESTED** [review]
Thank you. I like this! 👍

As discussed in slack, would you like to migrate the config-request into the subscriber and forward its value into the message?

**Comment by jeboehm** [comment]
Sorry, this escalated :D

File: src/Message/TinectRedirectUpdateHandler.php [hunk 1]
```diff
@@ -12,21 +14,38 @@
...
+    private function createOrUpdateRedirect(TinectRedirectUpdateMessage $message): string
     {
         $query = new RetryableQuery(
             $this->connection,
-            $this->connection->prepare('INSERT INTO `tinect_redirects_redirect` (`id`, `source`, `sales_channel_domain_id`, `created_at`)
+            $this->connection->prepare(
+                'INSERT INTO `tinect_redirects_redirect` (`id`, `source`, `sales_channel_domain_id`, `created_at`)
                       VALUES (:id, :source, :salesChannelDomainId, :createdAt)
-                      ON DUPLICATE KEY UPDATE `count` = count + 1')
+                      ON DUPLICATE KEY UPDATE `count` = count + 1, `updated_at` = NOW()'
```
Comment (tinect) [inline src/Message/TinectRedirectUpdateHandler.php:43]: nice catch 👍

File: src/Subscriber/ExceptionSubscriber.php [hunk 2]
```diff
@@ -89,37 +91,33 @@ private function handleRequest(Request $request): ?Response
...
+            userAgent: $request->headers->get('User-Agent') ?? '',
+            createRedirect: $this->canCreateRedirect($salesChannelDomainId),
+            id: $redirect?->getId(),
+        );
 
-        $message->setId($redirect->getId());
         $this->messageBus->dispatch($message);
 
-        if (!$redirect->isActive()) {
+        if ($redirect === null || !$redirect->isActive()) {
```
Comment (tinect) [inline src/Subscriber/ExceptionSubscriber.php:120]: Thank you for reducing the cognitive complexity

**Review by tinect — APPROVED** [review]

**Comment by tinect** [comment]
Thank you! :-)

**Pull request merged**